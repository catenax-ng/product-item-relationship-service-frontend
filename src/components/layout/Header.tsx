import styled from "@emotion/styled";
import { Box } from "@mui/material";
import { MainNavigation } from "cx-portal-shared-components";
import React from "react";
import { serverConfig } from "../../constants/serverConfig";
import { getCurrentEnvironment } from "../../utils/sessionStorageHandling";
import { HelpButton } from "./HelpButton";
import { LogOutButton } from "./LogOutButton";

type LinkItem = Partial<Record<"href" | "to", string>>;

export interface Tree {
  name: string;
  hint?: string;
  children?: Tree[];
}

export interface MenuItem extends LinkItem, Tree {
  title: string;
  children?: MenuItem[];
}

export const Header: React.FC = () => {
  const serverEnv = getCurrentEnvironment();

  return (
    <MainNavigation
      items={[
        {
          href: `/${serverEnv}/dashboard`,
          title: `${serverConfig[serverEnv].label} - Home`,
        },
      ]}
    >
      <Box
        component="img"
        src="/img/IRS_FE_Logo_long.png"
        sx={{
          display: "inline-block",
          height: "40px",
        }}
      />

      <Box>
        <HelpButton />
        <LogOutButton />
      </Box>
    </MainNavigation>
  );
};

export const PublicHeader: React.FC = () => {
  return (
    <StyledHeader>
      <MainNavigation items={[]}>
        <Box
          component="img"
          src="/img/cx-logo-text.svg"
          sx={{
            display: "inline-block",
            height: "40px",
            width: "170px",
          }}
        />
        <Box>
          <HelpButton />
        </Box>
      </MainNavigation>
    </StyledHeader>
  );
};

const StyledHeader = styled.header`
  position: relative;
  z-index: 10;
  align-items: center;
  justify-content: space-between;
  .d-flex {
    display: flex;
  }
`;
