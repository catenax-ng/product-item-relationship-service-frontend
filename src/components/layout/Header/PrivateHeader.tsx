import styled from "@emotion/styled";
import { MainNavigation } from "cx-portal-shared-components";
import React from "react";
import { useTranslation } from "react-i18next";
import { serverConfig } from "../../../utils/serverConfig";
import { getCurrentEnvironment } from "../../../utils/sessionStorageHandling";
import { HelpButton } from "./Buttons/HelpButton";
import { UserMenuButton } from "./Buttons/UserMenuButton";
import { LogoCatenaX } from "./Logo/LogoCatenaX";

/**
 * Private Header
 * Is the header used when the user is logged in
 *
 * @returns React component
 */
export const PrivateHeader: React.FC = () => {
  const serverEnv = getCurrentEnvironment();
  const { t } = useTranslation();

  return (
    <MainNavigation
      items={[
        {
          href: `/${serverEnv}/dashboard`,
          title: `${serverConfig(serverEnv).label} - ${t("content.irs.header.home")}`,
        },
      ]}
    >
      <LogoCatenaX />

      <StyledDiv>
        <HelpButton />
        <UserMenuButton />
      </StyledDiv>
    </MainNavigation>
  );
};

const StyledDiv = styled.div`
  display: flex;
`;
