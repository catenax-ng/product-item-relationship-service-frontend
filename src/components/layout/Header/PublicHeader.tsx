import styled from "@emotion/styled";
import { Box } from "@mui/material";
import { MainNavigation } from "cx-portal-shared-components";
import React from "react";
import { HelpButton } from "./Buttons/HelpButton";
import { LogoIRS } from "./Logo/LogoIRS";

/**
 * Public Header
 * Used only on the Welcome page
 *
 * @returns React component
 */
export const PublicHeader: React.FC = () => {
  return (
    <StyledHeader>
      <MainNavigation items={[]}>
        <LogoIRS />
        <Box>
          <HelpButton />
        </Box>
      </MainNavigation>
    </StyledHeader>
  );
};

/**
 * Additional Styles needed to ensure that the header is visible
 */
const StyledHeader = styled.header`
  position: relative;
  z-index: 10;
  align-items: center;
  justify-content: space-between;
  .d-flex {
    display: flex;
  }
`;
