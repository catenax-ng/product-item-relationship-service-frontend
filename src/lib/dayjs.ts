import dayjs from "dayjs";

export const defaultDateFormat = (value: string | Date) => dayjs(value).format("YYYY-MM-DD HH:mm:ss");
