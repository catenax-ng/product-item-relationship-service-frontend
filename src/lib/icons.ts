import ArrowForwardIcon from "@mui/icons-material/ArrowForward";
import FullscreenIcon from "@mui/icons-material/Fullscreen";
import FullscreenExitIcon from "@mui/icons-material/FullscreenExit";
import ErrorOutlineIcon from "@mui/icons-material/ErrorOutline";
import CancelIcon from "@mui/icons-material/Cancel";
export { ArrowForwardIcon, FullscreenExitIcon, FullscreenIcon, ErrorOutlineIcon, CancelIcon };
