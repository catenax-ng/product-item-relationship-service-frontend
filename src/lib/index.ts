export { Switch } from "@mui/material";
export type { SwitchProps } from "@mui/material";
export { styled } from "@mui/material/styles";
export * from "./icons";
export * from "./react-18next";
export * from "./react-query";
