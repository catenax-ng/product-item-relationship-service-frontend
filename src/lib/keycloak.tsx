import { ReactKeycloakProvider } from "@react-keycloak/web";
import Keycloak from "keycloak-js";
import React from "react";
import { serverConfig } from "../utils/serverConfig";
import { useServerEnv } from "../utils/ServerEnv";

const keycloakConfig: Keycloak.KeycloakConfig = {
  url: "https://centralidp.int.demo.catena-x.net/auth",
  realm: "CX-Central",
  clientId: "Cl2-CX-Portal",
};

export let keycloak = new Keycloak(keycloakConfig);

export const KeyCloakProvider: React.FC<{ children: React.ReactNode }> = ({ children }) => {
  const { serverEnv } = useServerEnv();
  //NOTE: This is needed as authURL is only for the login page, and not for the token auth provider
  keycloak = new Keycloak({ ...keycloakConfig, url: serverConfig(serverEnv).authServerUrl });
  return (
    <ReactKeycloakProvider authClient={keycloak} initOptions={{ pkceMethod: "S256" }}>
      {children}
    </ReactKeycloakProvider>
  );
};
