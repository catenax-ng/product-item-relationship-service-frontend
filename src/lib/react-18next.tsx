export { useTranslation } from "react-i18next";

import i18n, { changeLanguage } from "i18next";
import LanguageDetector from "i18next-browser-languagedetector";
import { initReactI18next } from "react-i18next";
import appsDE from "../assets/locales/de/apps.json";
import footerDE from "../assets/locales/de/footer.json";
import mainDE from "../assets/locales/de/main.json";
import notificationDE from "../assets/locales/de/notification.json";
import appsEN from "../assets/locales/en/apps.json";
import footerEN from "../assets/locales/en/footer.json";
import mainEN from "../assets/locales/en/main.json";
import notificationEN from "../assets/locales/en/notification.json";

const resources = {
  de: {
    translation: mainDE,
    footer: footerDE,
    apps: appsDE,
    notification: notificationDE,
  },
  en: {
    translation: mainEN,
    footer: footerEN,
    apps: appsEN,
    notification: notificationEN,
  },
};

const supportedLanguages = Object.keys(resources).sort();

const init = (): void => {
  i18n
    .use(LanguageDetector)
    .use(initReactI18next)
    .init({
      resources,
      fallbackLng: "en",
      supportedLngs: ["de", "en"],
      interpolation: {
        escapeValue: false,
      },
    })
    .catch((e) => console.error("Translation library init got error:", e));
};

export const I18nService = {
  init,
  changeLanguage,
  supportedLanguages,
};
