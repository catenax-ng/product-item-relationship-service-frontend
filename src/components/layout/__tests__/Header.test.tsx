import { KeyCloakProvider } from "../../../lib/keycloak";
import { renderWithProviders } from "../../../testing/test-utils";
import { PrivateHeader } from "../Header/PrivateHeader";
import { PublicHeader } from "../Header/PublicHeader";

it("renders a public header", () => {
  const { container } = renderWithProviders(
    <KeyCloakProvider>
      <PrivateHeader />
    </KeyCloakProvider>,
  );
  expect(container).toMatchSnapshot();
});

it("renders a private header", () => {
  const { container } = renderWithProviders(<PublicHeader />);
  expect(container).toMatchSnapshot();
});
