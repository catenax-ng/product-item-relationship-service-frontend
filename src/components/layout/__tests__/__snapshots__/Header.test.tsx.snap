// Vitest Snapshot v1

exports[`renders a private header 1`] = `
<div>
  <header
    class="css-on3ivn"
  >
    <div
      class="MuiBox-root css-1becb2a"
    >
      <div
        class="MuiBox-root css-429xfk"
      >
        <img
          class="MuiBox-root css-1emgxcc"
          src="/img/cx-logo-text.svg"
        />
      </div>
      <div
        class="MuiBox-root css-14tu3sa"
      >
        <nav
          class="MuiBox-root css-2imjyh"
        />
      </div>
      <div
        class="MuiBox-root css-19g4w2v"
      >
        <div
          class="MuiBox-root css-0"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeSmall MuiButton-containedSizeSmall css-73kr90-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            Help
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </header>
</div>
`;

exports[`renders a public header 1`] = `
<div>
  <div
    class="MuiBox-root css-1becb2a"
  >
    <div
      class="MuiBox-root css-429xfk"
    >
      <img
        class="MuiBox-root css-eerkt0"
        src="/img/IRS_FE_Logo_long.png"
      />
    </div>
    <div
      class="MuiBox-root css-14tu3sa"
    >
      <nav
        class="MuiBox-root css-2imjyh"
      >
        <div
          class="MuiBox-root css-79elbk"
        >
          <a
            class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-40uvlr-MuiTypography-root-MuiLink-root-MuiTypography-root-MuiLink-root"
            href="/DEMO/dashboard"
          >
            Demo Data - Home
          </a>
        </div>
      </nav>
    </div>
    <div
      class="MuiBox-root css-19g4w2v"
    >
      <div
        class="MuiBox-root css-0"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeSmall MuiButton-containedSizeSmall css-73kr90-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          Help
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <button
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary MuiIconButton-sizeMedium css-1kuq5xv-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="LogoutIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="m17 7-1.41 1.41L18.17 11H8v2h10.17l-2.58 2.58L17 17l5-5zM4 5h8V3H4c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h8v-2H4V5z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;
